Add platformList

https://github.com/saagie/gradle-saagie-dataops-plugin/wiki/platformList

tested in 1.1.30 :

➜  test plugin gradle dataops gradle platformList
> Task :platformList FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':platformList'.
> Cannot invoke method toUpperCase() on null object

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 954ms
1 actionable task: 1 executed

Ok i've update the doc and my test.

Not better : 

➜  test plugin gradle dataops gradle platformList

> Task :platformList FAILED
Missing params in plugin configuration: https://github.com/saagie/gradle-saagie-dataops-plugin/wiki/projectsList

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':platformList'.
> Missing params in plugin configuration: https://github.com/saagie/gradle-saagie-dataops-plugin/wiki/projectsList

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 957ms
1 actionable task: 1 executed

**===> the help link is not good.**

 server {
        url = 'https://saagie-beta.prod.saagie.io'
        login = 'youen.chene'
        password = xxx
        jwt = true
        realm = 'saagie'
    }

--- a/src/main/groovy/io/saagie/plugin/dataops/models/Server.groovy
+++ b/src/main/groovy/io/saagie/plugin/dataops/models/Server.groovy
@@ -15,7 +15,6 @@ class Server {
     @NotNull(message = 'password cannot be null')
     String password
 
-    @NotBlank(message = 'environment cannot be empty')
     String environment
 
     String proxyHost = null

--- a/src/test/groovy/io/saagie/plugin/DataOpsPluginTest.groovy
+++ b/src/test/groovy/io/saagie/plugin/DataOpsPluginTest.groovy
@@ -102,7 +102,6 @@ class DataOpsPluginTest extends Specification {
         then:
         Exception e = thrown()
         result == null
-        e.message.contains('environment cannot be empty')
         e.message.contains('url cannot be empty')
         e.message.contains('Missing required params in plugin configuration, check that you have url, environment, login and password defined in your server object.')
     }

--- a/src/test/groovy/io/saagie/plugin/platform/PlatformListTaskTests.groovy
+++ b/src/test/groovy/io/saagie/plugin/platform/PlatformListTaskTests.groovy
@@ -79,7 +79,6 @@ class PlatformListTaskTests extends Specification {
                     url = 'http://localhost:9000'
                     login = 'fake.user'
                     password = 'ThisPasswordIsWrong'
-                    environment = 2
                     realm = 'saagie'
                     jwt = true
                 }
@@ -102,7 +101,6 @@ class PlatformListTaskTests extends Specification {
                     url = 'http://localhost:9000'
                     login = 'fake.user'
                     password = 'ThisPasswordIsWrong'
-                    environment = 2
                 }
             }
         '''

